packages/material-ui-styles/src/index.d.ts

```diff
@@ -82,6 +82,40 @@ declare module '@material-ui/styles/makeStyles' {
 
   // https://stackoverflow.com/a/49928360/3406963 without generic branch types
   type IsAny<T> = 0 extends (1 & T) ? true : false;
+  /**
+   * @internal
+   *
+   * check if a type is `{}`
+   *
+   * 1. false if the given type has any members
+   * 2. false if the type is `object` which is the only other type with no members
+   *  {} is a top type so e.g. `string extends {}` but not `string extends object`
+   */
+  export type IsEmptyInterface<T> = keyof T extends never
+    ? string extends T
+      ? true
+      : false
+    : false;
+
+  type Or<A, B> = A extends true ? true : B extends true ? true : false;
```

Comment:
Definitely. I would defer that to a different PR though since we have plenty of unrelated candidates for this (`Pick`, `IsAny`?) and I want to to have a look at different options. I for one only used `utility-types`.